### DISCUSSION TITLE
Parsing and extending Envelopes data from clients

### DISCUSSION THREAD

**Discussion by ktsangop** [getsentry/sentry-java#2531] [Q&A]
Hello everyone!

As discussed on the client side SDK thread here
https://github.com/getsentry/sentry-javascript/discussions/7112#discussioncomment-4932884

We would like any information on how to parse Envelopes sent from sentry-javascript clients, to our private API
that's implemented using Java/Spring.
Also we need to extend Envelopes client data with more information that our server has about user session etc.

Thank you in advance!

**Comment by adinauer** [comment]
Hello @ktsangop, maybe taking a look at https://github.com/getsentry/sentry-java/blob/6615b75f03a0818db8e35b196ef6d337bb8bf97e/sentry/src/main/java/io/sentry/OutboxSender.java helps you.

**Reply by Smaragda2** [reply]
This seems to read a file and then create/send the SentryEnvelope. 
But is there a way to convert, at the sentry-java the non-valid JSON format that is generated from the sentry-javascript client, and then sent it to the POST /api/<project_id>/envelope/?

**Reply by adinauer** [reply]
Sorry, I'm not terribly familiar with specifics about the JS SDK but the sample I linked above reads from a file that is generated by other Sentry SDKs so should work in theory. Have you tried it and are seeing specific errors?
